rdb/src/main/java/org/openmbee/mms/rdb/repositories/BaseDAOImpl.java

```diff
@@ -38,7 +40,7 @@ public JdbcTemplate getConn() {
 
     public String getSuffix() {
         String refId = ContextHolder.getContext().getBranchId();
-        if(Pattern.matches("^[a-zA-Z0-9-_]*$", refId)) { 
+        if(Pattern.matches(BRANCH_ID_VALID_PATTERN, refId)) {
```

Comment:
From what I understand Pattern.matches(String, String) is a convenience method that does Pattern.matches(Pattern.compile(String), String) for regex patterns that are used only once. The expensive bit is Pattern.compile, which should instead be defined as static final so that it is only compiled once at initialization.

Reply:
Changed to a static Pattern instead